### PR TITLE
[lldb] Revise IDE folder structure

### PR DESCRIPTION
**Pull request by Meinersbur**
Reviewers of #89153 suggested to break up the patch into per-subproject patches. This is the LLDB part. See #89153 for the entire series and motivation.

Update the folder titles for targets in the monorepository that have not seen taken care of for some time. These are the folders that targets are organized in Visual Studio and XCode (`set_property(TARGET <target> PROPERTY FOLDER "<title>")`) when using the respective CMake's IDE generator.

 * Ensure that every target is in a folder
 * Use a folder hierarchy with each LLVM subproject as a top-level folder
 * Use consistent folder names between subprojects
 * When using target-creating functions from AddLLVM.cmake, automatically deduce the folder. This reduces the number of `set_property`/`set_target_property`, but are still necessary when `add_custom_target`, `add_executable`, `add_library`, etc. are used. A LLVM_SUBPROJECT_TITLE definition is used for that in each subproject's root CMakeLists.txt.

### CODE REVIEW

File: lldb/source/API/CMakeLists.txt [hunk 1]
```diff
@@ -232,7 +231,7 @@ elseif (LLDB_EXPORT_ALL_SYMBOLS)
   )
 
   add_llvm_symbol_exports(liblldb ${exported_symbol_file})
-  set_target_properties(liblldb_exports PROPERTIES FOLDER "lldb misc")
+  set_target_properties(liblldb PROPERTIES FOLDER "LLDB/Misc")
```
Comment (JDevlieghere) [inline lldb/source/API/CMakeLists.txt:234]: Should this be `LLDB/API`?

**Review by JDevlieghere — APPROVED** [review]
Thanks for splitting this up. LGTM modulo one folder name.

File: lldb/source/API/CMakeLists.txt (hunk 1 above)
Comment (Meinersbur) [reply lldb/source/API/CMakeLists.txt:234]: This `set_target_properties` is actually redundant: `add_llvm_symbol_exports` already puts it into a folder. It is also `${LLVM_SUBPROJECT_TITLE}/Misc`. I agree `${LLVM_SUBPROJECT_TITLE}/API` would be a better name. I am going to make that change in #89741. Thanks.

**Review by JDevlieghere — APPROVED** [review]

**Review by bulbazord — APPROVED** [review]

**Review by medismailben — APPROVED** [review]
Nice! LGTM!

**Pull request merged**